File: frontend/src/admin/data/roles.js
// UI-only mirror of the backend's role -> sections map (app/core/roles.py).
// This just decides what renders in the sidebar. It is NOT access control —
// that's enforced server-side, same split SACHI uses (see require_role()).
//
// Until the DB-backed admin/role backend lands, useAdminAuth's `role` is
// null — SECTIONS_BY_ROLE isn't consulted in that case and every section
// shows, so the app keeps working unchanged in the meantime.
export const SECTIONS_BY_ROLE = {
  superadmin: ["clients", "reviews", "invoices", "admins"],
  admin: ["clients", "reviews", "invoices", "admins"], // "admins" renders read-only for this role
  invoice_admin: ["invoices"],
};

export function sectionsForRole(role) {
  if (!role) return null; // null = "show everything" (pre-RBAC fallback)
  return SECTIONS_BY_ROLE[role] || [];
}

export function canManageAdmins(role) {
  return role === "superadmin";
}

export const ROLE_LABELS = {
  superadmin: "Super Admin",
  admin: "Admin",
  invoice_admin: "Invoice Admin",
};

File: backend/app/core/roles.py
"""Server-side source of truth for role -> section access. Mirrors SACHI's
app/models/common.py pattern. Keep in sync by hand with
frontend/src/admin/data/roles.js (that copy is UI-only; this one is what's
actually enforced).

"admins" access is deliberately split: everyone in ROLES who has "admins"
can GET the list, but only "superadmin" can create/delete/reset — that's
enforced in the admin_users router itself (require_superadmin on those
three routes), not by this map. This map only answers "can this role see
the section at all".
"""

ROLES: dict[str, list[str]] = {
    "superadmin": ["clients", "reviews", "invoices", "admins"],
    "admin": ["clients", "reviews", "invoices", "admins"],
    "invoice_admin": ["invoices"],
}

ROLE_LABELS: dict[str, str] = {
    "superadmin": "Super Admin",
    "admin": "Admin",
    "invoice_admin": "Invoice Admin",
}


def role_has_section(role: str, section: str) -> bool:
    return section in ROLES.get(role, [])
